R/bioc-standalone.R

```diff
@@ -200,10 +201,24 @@ bioconductor <- local({
     if (minor %in% names(builtin_map)) return(builtin_map[[minor]])
 
     # If we are not in the map, then we need to look this up in
-    # YAML data.
+    # YAML data. It is possible that the current R version matches multiple
+    # Bioc versions. Then we choose the latest released version. If none
+    # of them were released (e.g. they are 'devel' and 'future'), then
+    # we'll use the 'devel' version.
 
     map <- get_version_map(forget = forget)
-    mine <- match(package_version(minor), map$r_version)
+    mine <- which(package_version(minor) == map$r_version)
+    if (length(mine) == 0) {
+      mine <- NA
+    } else if (length(mine) > 1) {
+      if ("release" %in% map$bioc_status[mine]) {
+        mine <- mine["release" == map$bioc_status[mine]]
+      } else if ("devel" %in% map$bioc_status[mine]) {
+        mine <- mine["devel" == map$bioc_status[mine]]
+      } else {
+        mine <- rev(mine)[1]
```

Comment:
Should this be `[[1]]`?

Reply:
Also maybe using `tail(mine, n = 1)` is possibly more clear?

Reply:
No, it has to be `[1]` because we need an `NA` if there is no match, instead of an error.

We cannot use `tail()` because this is a standalone file, it should not import the utils package, ideally. Although we already load it for `download.file`, but still in general we should avoid non-base functions.

Reply:
Ok, but aren't we handling the no match case with `length(mine) == 0` above?